curation-migration-datasync/eventConsumer.integration.ts

```diff
@@ -335,6 +336,22 @@ describe('event consumption integration test', function () {
       );
       expect(dymamoDbSpy.callCount).toEqual(0);
     });
+
+    it('should not process duplicate events', async () => {
+      const consoleSpy = sinon.spy(console, 'log');
+
+      //item already present in dynamo
+      const duplicateEvent = {
+        ...testEventBody,
+        scheduledItemExternalId: curatedItemRecords[0].scheduledItemExternalId,
+      };
+
+      await addScheduledItem(duplicateEvent, db);
+      expect(consoleSpy.calledOnce).toBe(true);
+      expect(consoleSpy.getCall(0).firstArg).toContain(
+        'duplicate add-scheduled-item event'
+      );
```

Comment:
not a blocker - but do you think it's worth making sure `curatedItemModel.upsertFromEvent` was _not_ called?

might be good to guard against future regressions?

Reply:
good point, i couldn't spy on `curatedItemModel` and `dbService` as the objects were created locally inside the function, might require refactoring function definition and tests

so spied on `db` objects and made sure we are never calling the `db.transaction` function to make sure db write doesn't happen